java/src/main/java/ai/rapids/cudf/DType.java

```diff
@@ -15,78 +15,292 @@
+  public static final DType INT64 = new DType(DTypeEnum.INT64);
+  public static final DType UINT8 = new DType(DTypeEnum.UINT8);
+  public static final DType UINT16 = new DType(DTypeEnum.UINT16);
+  public static final DType UINT32 = new DType(DTypeEnum.UINT32);
+  public static final DType UINT64 = new DType(DTypeEnum.UINT64);
+  public static final DType FLOAT32 = new DType(DTypeEnum.FLOAT32);
+  public static final DType FLOAT64 = new DType(DTypeEnum.FLOAT64);
+  public static final DType BOOL8 = new DType(DTypeEnum.BOOL8);
+  public static final DType TIMESTAMP_DAYS = new DType(DTypeEnum.TIMESTAMP_DAYS);
+  public static final DType TIMESTAMP_SECONDS = new DType(DTypeEnum.TIMESTAMP_SECONDS);
+  public static final DType TIMESTAMP_MILLISECONDS = new DType(DTypeEnum.TIMESTAMP_MILLISECONDS);
+  public static final DType TIMESTAMP_MICROSECONDS = new DType(DTypeEnum.TIMESTAMP_MICROSECONDS);
+  public static final DType TIMESTAMP_NANOSECONDS = new DType(DTypeEnum.TIMESTAMP_NANOSECONDS);
+  public static final DType DURATION_DAYS = new DType(DTypeEnum.DURATION_DAYS);
+  public static final DType DURATION_SECONDS = new DType(DTypeEnum.DURATION_SECONDS);
+  public static final DType DURATION_MILLISECONDS = new DType(DTypeEnum.DURATION_MILLISECONDS);
+  public static final DType DURATION_MICROSECONDS = new DType(DTypeEnum.DURATION_MICROSECONDS);
+  public static final DType DURATION_NANOSECONDS = new DType(DTypeEnum.DURATION_NANOSECONDS);
+  public static final DType STRING = new DType(DTypeEnum.STRING);
+  public static final DType LIST = new DType(DTypeEnum.LIST);
+  public static final DType STRUCT = new DType(DTypeEnum.STRUCT);
+
+  /* This is used in fromNative method to return singleton object for non-decimal types.
+  Please make sure the order here is same as that of DTypeEnum. Whenever a new non-decimal
+  type is added in DTypeEnum, this array needs to be updated as well.*/
+  private static final DType[] SINGLETON_DTYPE_LOOKUP = new DType[]{
+      EMPTY,
+      INT8,
+      INT16,
+      INT32,
+      INT64,
+      UINT8,
+      UINT16,
+      UINT32,
+      UINT64,
+      FLOAT32,
+      FLOAT64,
+      BOOL8,
+      TIMESTAMP_DAYS,
+      TIMESTAMP_SECONDS,
+      TIMESTAMP_MILLISECONDS,
+      TIMESTAMP_MICROSECONDS,
+      TIMESTAMP_NANOSECONDS,
+      DURATION_DAYS,
+      DURATION_SECONDS,
+      DURATION_MILLISECONDS,
+      DURATION_MICROSECONDS,
+      DURATION_NANOSECONDS,
+      null, // DICTIONARY32
+      STRING,
+      LIST,
+      null, // DECIMAL32
+      null, // DECIMAL64
+      STRUCT
+  };
+
   /**
-   * s since the UNIX epoch
+   * This only works for fixed width types. Variable width types like strings the value is
+   * undefined and should be ignored.
+   *
+   * @return size of type in bytes.
    */
-  TIMESTAMP_SECONDS(8, 13, "timestamp[s]"),
+  public int getSizeInBytes() { return typeId.sizeInBytes; }
+
   /**
-   * ms since the UNIX epoch
+   * Returns scale for Decimal Type
```

Comment:
This should explain what scale is since many will find it unintuitive compared to Java's `BigDecimal`.